app/config/ServiceConfiguration.scala

```diff
@@ -63,25 +64,57 @@ class PlayBasedServiceConfiguration @Inject()(configuration: Configuration, env:
   override def retryInterval = getRequired(configuration.getMilliseconds, "aws.sqs.retry.interval").milliseconds
 
   override def s3UrlExpirationPeriod(serviceName: String): FiniteDuration = {
-    val urlExpirationConfig =
-      s"${runMode.env}.upscan.consuming-services.${replaceInvalidJsonChars(serviceName)}.aws.s3.urlExpirationPeriod"
-    val expirationDuration: FiniteDuration =
-      getRequired(configuration.getMilliseconds, urlExpirationConfig).milliseconds
-
-    if (expirationDuration <= 7.days) {
-      expirationDuration
-    } else {
-      Logger.warn(
-        s"Expiration period for key: [$urlExpirationConfig] was: [$expirationDuration]. Using maximum value of 7 days instead.")
-      1.day
+    val configKey = configKeyForConsumingService(serviceName, S3UrlExpirationPeriod.ConfigDescriptor)
+    val maybeErrorOrExpiry = configuration.getMilliseconds(configKey).map(_.milliseconds).map(validateS3UrlExpirationPeriod(_,
+      msgWhenInvalid = s"Invalid configuration found for [${S3UrlExpirationPeriod.ConfigDescriptor}] for service [$serviceName] - trying default")
+    )
+    if (maybeErrorOrExpiry.isEmpty) Logger.debug(s"No configuration found for [${S3UrlExpirationPeriod.ConfigDescriptor}] for service [$serviceName] - trying default")
+    maybeErrorOrExpiry.foreach(_.left.foreach(msg => Logger.warn(msg)))
+
+    val expirationPeriod = maybeErrorOrExpiry.fold(ifEmpty = defaultS3UrlExpirationPeriod()) {
+      _.fold(_ => defaultS3UrlExpirationPeriod(), identity)
     }
+    Logger.debug(s"Using configuration for [${S3UrlExpirationPeriod.ConfigDescriptor}] for service [$serviceName] of [$expirationPeriod]")
+    expirationPeriod
   }
 
   override def endToEndProcessingThreshold(): Duration =
     getRequired(configuration.getMilliseconds, "upscan.endToEndProcessing.threshold").seconds
 
-  def getRequired[T](function: String => Option[T], key: String) =
+  private def getRequired[T](function: String => Option[T], key: String): T =
     function(key).getOrElse(throw new IllegalStateException(s"Configuration key not found: $key"))
-  private[config] def replaceInvalidJsonChars(serviceName: String): String =
+
+  private def replaceInvalidJsonChars(serviceName: String): String =
     serviceName.replaceAll("[/.]", "-")
+
+  private def defaultS3UrlExpirationPeriod(): FiniteDuration = {
+    val errorOrExpiry = configuration.getMilliseconds(configKeyForDefault(S3UrlExpirationPeriod.ConfigDescriptor)).map(_.milliseconds).toRight(
```

Comment:
This part, up until the `.right.flatMap(validateS3UrlExpirationPeriod(_` seems almost identical to the part in s3UrlExpirationPeriod but uses Either instead of `Option[Either]` ... is there something in common there that can be extracted to make the error handling easier?